bump.lua

```diff
@@ -28,6 +28,39 @@ local bump = {
   ]]
 }
 
```

Comment:
So, the first change here is to add a table pool so that we can reuse temporary tables instead of throwing them away. This is used for a few operations:

- The table used to keep track of visited items in `World:project()`.
- The table used to store the results of `getDictItemsInCellRect`
- A new table used to keep track of visited items during `World:projectMove()` (formerly `World:check()`). More on this later.